java-checks/src/main/java/org/sonar/java/checks/tests/MockitoArgumentMatchersUsedOnAllParametersCheck.java

```diff
@@ -52,25 +52,25 @@
 
   @Override
   protected void visitArguments(Arguments arguments) {
-    List<Tree> argumentMatchers = new ArrayList<>();
-    List<Tree> secondaries = new ArrayList<>();
-
+    if (arguments.isEmpty()) {
+      return;
+    }
+    List<Tree> nonMatchers = new ArrayList<>();
     for (ExpressionTree arg : arguments) {
       arg = ExpressionUtils.skipParentheses(arg);
-      if (isArgumentMatcherLike(arg) || isCastArgumentMatcherLike(arg)) {
-        argumentMatchers.add(arg);
-      } else {
-        secondaries.add(arg);
+      if (!isArgumentMatcherLike(arg)) {
+        nonMatchers.add(arg);
       }
     }
 
-    if (argumentMatchers.isEmpty()) {
-      return;
-    }
-    if (argumentMatchers.size() < arguments.size()) {
-      reportIssue(secondaries.get(0),
-        "Add an \"eq()\" argument matcher on this/these parameters",
-        secondaries.stream()
+    if (0 < nonMatchers.size() && nonMatchers.size() < arguments.size()) {
```

Comment:
You can extract the `size()` call to a variable (used later aswell) and use `isEmpty()` here.